manim/mobject/types/vectorized_mobject.py

```diff
@@ -1017,6 +1041,50 @@ def point_from_proportion(self, alpha: float) -> np.ndarray:
 
             current_length += length
 
+    def proportion_from_point(
+        self,
+        point: typing.Iterable[typing.Union[float, int]],
+    ) -> float:
+        """Returns the proportion along the path of the :class:`VMobject`
+        a particular given point is at.
+
+        Parameters
+        ----------
+        point
+            The Cartesian coordinates of the point which may or may not lie on the :class:`VMobject`
+
+        Returns
+        -------
+        float
+            The proportion along the path of the :class:`VMobject`.
+
+        Raises
+        ------
+        :exc:`ValueError`
+            If ``point`` does not lie on the curve.
+        :exc:`Exception`
+            If the :class:`VMobject` has no points.
+        """
+        self.throw_error_if_no_points()
+
```

Comment:
It would be good to give some paragraph-level overview of what you're doing here. If I'm reading this correctly, you're iterating over segments, keeping track of how far you've gone, and once you find the matching point, you find how far it is along that curve, then use the total to get the proportion.

Reply:
Done.